src/commands/InspectCommand.ts

```diff
@@ -0,0 +1,58 @@
+import {GuildMember, Message, MessageEmbed} from "discord.js";
+import DiscordUtil from "../utils/DiscordUtil";
+import Command from "../abstracts/Command";
+import DateUtils from "../utils/DateUtils";
+import { EMBED_COLOURS } from "../config.json";
+
+class InspectCommand extends Command {
+	constructor() {
+		super(
+			"inspect",
+			"Show information about a given user"
+		);
+	}
+
+	async run(message: Message, args: string[]) {
+		const userObj = args.length > 0
+			? await DiscordUtil.getGuildMember(args[0], message.guild!)
+			: message.member!;
+
+		const embed = userObj === undefined
+			? this.buildNoMatchEmbed()
+			: this.buildInspectEmbed(userObj!);
+
+		await message.channel.send({embed});
+	}
+
+	private buildNoMatchEmbed(): MessageEmbed {
+		const embed = new MessageEmbed();
+
+		embed.setTitle("Error");
+		embed.setDescription("No match found.");
+		embed.addField("Correct Usage", "?inspect [username|userID]");
+		embed.setColor(EMBED_COLOURS.ERROR);
+		return embed;
+	}
+
+	private buildInspectEmbed(memberObj: GuildMember): MessageEmbed {
+		const embed = new MessageEmbed();
+
+		embed.setTitle(`Inspecting ${memberObj?.user.tag}`);
+		embed.setColor(memberObj?.displayColor || EMBED_COLOURS.DEFAULT);
+		embed.setThumbnail(memberObj?.user.displayAvatarURL());
+		embed.addField("User ID", memberObj?.user.id);
+		embed.addField("Username", memberObj?.user.tag);
+		if (memberObj?.nickname !== null) embed.addField("Nickname", memberObj?.nickname);
+		if (memberObj?.nickname !== null) embed.addField("Joined At", DateUtils.formatAsText(memberObj?.joinedAt!));
```

Comment:
This is questionable, if nickname exists show the join date?
Also the ? in here are not needed, as you are now sure that `memeberObj` exists, this is for the whole buildInspectEmbed function

So it could be
```ts
if (memberObj.joinedAt) embed.addField("Joined At", DateUtils.formatAsText(memberObj.joinedAt));
```

Reply:
Fixed